app/FileJson.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace ProjektGopher\Whisky;
+
+use Illuminate\Support\Facades\File;
+use Swaggest\JsonSchema\Context;
+use Swaggest\JsonSchema\Schema;
+
+class FileJson
+{
+    public string $path;
+
+    public function __construct(
+        public string $filePath,
+    ) {
+        $this->path = $filePath;
+    }
+
+    public static function make(string $filePath): FileJson
+    {
+        return new FileJson($filePath);
+    }
+
+    public function read(bool $validate = true): string|array|null
+    {
+        $content = json_decode(File::get($this->path), true);
+
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            $msg = 'Invalid JSON';
+            if (function_exists('json_last_error_msg')) {
+                $msg .= ': '.json_last_error_msg();
+                $msg .= ' in '.$this->path;
+            }
+            throw new \Exception($msg);
+        }
+
+        if ($validate) {
+            try {
+                $contentJson = json_decode(File::get($this->path));
+                $this->validate($contentJson);
+            } catch (\Exception $e) {
+                $msg = 'Invalid JSON schema';
+                $msg .= ': '.$e->getMessage();
+                $msg .= ' in '.$this->path;
+                throw new \Exception($msg);
+            }
+        }
+
+        return $content;
+    }
+
+    /**
+     * @param  \stdClass|array|string|int|float|bool|null  $content
+     *
+     * @throws \Exception
+     */
+    protected function validate(mixed $content): void
+    {
+        $options = new Context();
+        // $options->version = 7;
+
+        $schema = Schema::import($this->getSchemaValidation(), $options);
+
+        $schema->in($content);
+    }
+
+    protected function getSchemaValidation(): ?\stdClass
+    {
+        // return json_decode(File::get(Whisky::base_path('resources/schemas/whisky.json')), false);
```

Comment:
```suggestion
        
```

Reply:
I don't mind the idea of extracting this, so I'm going to leave this comment in as a reminder for now.